### PR TITLE
RDM-4523-To resolve conflicts of rebase

### PR DESCRIPTION
**Pull request by ganganna-peddapothu**
### Performance - improve performance of case type def retrieval: Complex Type, This is same PR as https://github.com/hmcts/ccd-definition-store-api/pull/378, created to resolve conflicts. ###
**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[X] No
```

### CODE REVIEW

File: domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/service/EntityToResponseDTOMapper.java [hunk 1]
```diff
@@ -126,6 +129,13 @@
                          + "       )",
             target = "fixedListItems"
     )
+    @Mapping(
+            expression = "java("
+                         + "           uk.gov.hmcts.ccd.definition.store.domain.service.EntityToResponseDTOMapper.ComplexFieldsMapper.map("
+                         + "               fieldTypeEntity))",
+            target = "complexFields"
+    )
+
```
Comment (mario-paniccia) [inline domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/service/EntityToResponseDTOMapper.java]: remove space please

File: domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/service/EntityToResponseDTOMapper.java [hunk 2]
```diff
@@ -256,4 +266,32 @@ private FixedListMapper() {
...
+            } else if (baseFieldTypeEntity != null && isComplexField(baseFieldTypeEntity.getReference())) {
+                return getCaseFields(fieldTypeEntity.getComplexFields());
+            }
+            return new ArrayList<>();
+        }
+
+        private static List<CaseField> getCaseFields(List<ComplexFieldEntity> complexFieldEntityList) {
+            List<CaseField> caseFields = new ArrayList<>();
+            for (ComplexFieldEntity complexFieldEntity : complexFieldEntityList) {
+                caseFields.add(EntityToResponseDTOMapper.INSTANCE.map(complexFieldEntity));
```
Comment (mario-paniccia) [inline domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/service/EntityToResponseDTOMapper.java]: could we refactor to use Streams.map method?

Comment (ganganna-peddapothu) [reply domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/service/EntityToResponseDTOMapper.java]: DONE

File: domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/service/EntityToResponseDTOMapper.java (hunk 1 above)
Comment (ganganna-peddapothu) [reply domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/service/EntityToResponseDTOMapper.java]: DONE

**Comment by Olawale** [comment]
Response times for the performance improvement fix are in the Jira ticket https://tools.hmcts.net/jira/browse/RDM-4523
Also carried out regression testing on case management web. No issues found.

File: domain/src/test/java/uk/gov/hmcts/ccd/definition/store/domain/service/EntityToResponseDTOMapperTest.java [hunk 3]
```diff
@@ -1010,18 +1027,49 @@ private FieldType mapAssertCommonFieldsAndReturn(FieldTypeEntity fieldTypeEntity
...
+
+            assertEquals(fieldTypeEntity.getListItems().size(), fieldType.getFixedListItems().size());
+            assertEquals(fieldType.getFixedListItems().get(0).getLabel(), fixedListItem1.getLabel());
+            assertEquals(fieldType.getFixedListItems().get(1).getLabel(), fixedListItem2.getLabel());
+            assertEquals(fieldType.getFixedListItems().get(2).getLabel(), fixedListItem3.getLabel());
+
+            assertEquals(fieldTypeEntity.getCollectionFieldType(), fieldType.getCollectionFieldType());
+
+            return fieldType;
+
```
Comment (tzarsmango) [inline domain/src/test/java/uk/gov/hmcts/ccd/definition/store/domain/service/EntityToResponseDTOMapperTest.java]: should we add a test for verifying that the INSTANCE mapper not being called for non-complex entity types?
I mean sth like `verify(spyOnClassUnderTest, never()).map(fieldTypeEntity)`

**Pull request merged**